lib/Controller/TagsController.php

```diff
@@ -80,6 +88,21 @@ public function update(int $id, string $displayName, string $color): JSONRespons
 		$tag = $this->mailManager->updateTag($id, $displayName, $color, $this->currentUserId);
 		return new JSONResponse($tag);
 	}
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @throws ClientException
+	 */
+	#[TrapError]
+	public function delete(int $id, int $accountId): JSONResponse {
+		try {
+			$account = $this->accountService->find($this->currentUserId, $accountId);
```

Comment:
The message tags are shared for a user and therefore delete tag should remove it for every account.